### PR TITLE
Add `delete_creating_tasks` option for `internal.free()`

### PR DESCRIPTION
**Pull request by jovany-wang**
## What do these changes do?

Add an option `delete_creating_tasks` for `free()` to indicate whether also delete the obkects' creating tasks.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: python/ray/internal/internal_api.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 __all__ = ["free"]
 
 
-def free(object_ids, local_only=False):
+def free(object_ids, local_only=False, delete_creating_tasks=False):
```
Comment (raulchen) [inline python/ray/internal/internal_api.py]: document this new arg

**Review by raulchen — COMMENTED** [review]
Could you also add a test?

File: src/ray/raylet/lib/java/org_ray_runtime_raylet_RayletClientImpl.h [hunk 2]
```diff
@@ -96,7 +96,7 @@ Java_org_ray_runtime_raylet_RayletClientImpl_nativeGenerateTaskId(JNIEnv *, jcla
 JNIEXPORT void JNICALL
 Java_org_ray_runtime_raylet_RayletClientImpl_nativeFreePlasmaObjects(JNIEnv *, jclass,
                                                                      jlong, jobjectArray,
-                                                                     jboolean);
+                                                                     jboolean, jboolean);
```
Comment (guoyuhong) [inline src/ray/raylet/lib/java/org_ray_runtime_raylet_RayletClientImpl.h]: Also update the comment of the function signature in both .h and .cc files?

**Review by raulchen — APPROVED** [review]

**Review by guoyuhong — APPROVED** [review]
LGTM.

**Pull request merged**